### PR TITLE
KAFKA-4641: Add more unit test for stream thread

### PR DESCRIPTION
**Pull request by guozhangwang**
Before the patch, jacoco coverage test:

Element | Missed Instructions | Cov. | Missed Branches | Cov. | Missed | Cxty | Missed | Lines | Missed | Methods | Missed | Classes
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
Total | 3,386 of 22,177 | 84% | 336 of 1,639 | 79% | 350 | 1,589 | 526 | 4,451 | 103 | 768 | 1 | 102
StreamThread |   | 77% |   | 76% | 27 | 102 | 48 | 299 | 1 | 31 | 0 | 1

After the patch:

Element | Missed Instructions | Cov. | Missed Branches | Cov. | Missed | Cxty | Missed | Lines | Missed | Methods | Missed | Classes
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
Total | 3,329 of 22,180 | 84% | 329 of 1,639 | 79% | 345 | 1,590 | 516 | 4,452 | 102 | 769 | 1 | 102
StreamThread |   | 81% |   | 80% | 23 | 103 | 39 | 300 | 1 | 32 | 0 | 1

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@bbejeck

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java [hunk 1]
```diff
@@ -759,6 +765,151 @@ public void shouldReturnStandbyTaskMetadataWhileRunningState() throws Interrupte
...
+        standbyTasks.put(task1, Collections.singleton(t1p1));
+        standbyTasks.put(task3, Collections.singleton(t2p1));
+
+        thread.taskManager().setAssignmentMetadata(Collections.<TaskId, Set<TopicPartition>>emptyMap(), standbyTasks);
+
+        thread.rebalanceListener.onPartitionsAssigned(Collections.<TopicPartition>emptyList());
+
+        thread.runOnce(-1);
+
+        StandbyTask standbyTask1 = thread.taskManager().standbyTask(partition1);
```
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java]: nit: final and also below

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java [hunk 2]
```diff
@@ -759,6 +765,151 @@ public void shouldReturnStandbyTaskMetadataWhileRunningState() throws Interrupte
         assertTrue(threadMetadata.activeTasks().isEmpty());
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldUpdateStandbyTask() throws InterruptedException {
```
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java]: do we need the `throws InterruptedException`? doesn't seem to be thrown anywhere

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java [hunk 3]
```diff
@@ -759,6 +765,151 @@ public void shouldReturnStandbyTaskMetadataWhileRunningState() throws Interrupte
...
+        assertEquals(10L, store1.approximateNumEntries());
+        assertEquals(5L, store2.approximateNumEntries());
+        assertEquals(Collections.singleton(partition2), restoreConsumer.paused());
+        assertEquals(1, thread.standbyRecords().size());
+        assertEquals(5, thread.standbyRecords().get(partition2).size());
+    }
+
+    @Test
+    public void shouldPunctuateActiveTask() throws InterruptedException {
+        final List<Long> punctuatedStreamTime = new ArrayList<>();
```
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java]: this test doesn't seem to throw `InterruptedException` as well

**Comment by bbejeck** [comment]
left some minor comments

**Review by dguy — APPROVED** [review]
LGTM

**Review by bbejeck — APPROVED** [review]
Thanks, @guozhangwang LGTM

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java [hunk 4]
```diff
@@ -759,6 +765,151 @@ public void shouldReturnStandbyTaskMetadataWhileRunningState() throws Interrupte
...
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldUpdateStandbyTask() {
+        final String storeName1 = "count-one";
+        final String storeName2 = "table-two";
+        final String changelogName = applicationId + "-" + storeName1 + "-changelog";
+        final TopicPartition partition1 = new TopicPartition(changelogName, 1);
+        final TopicPartition partition2 = t2p1;
+        internalStreamsBuilder.stream(Collections.singleton(topic1), consumed)
+                .groupByKey().count(Materialized.<Object, Long, KeyValueStore<Bytes, byte[]>>as("count-one"));
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java]: `"count-one"` -> `storeName1`

**Review by mjsax — APPROVED** [review]
One nit. Feel free to merge after addressed.

**Pull request merged**